src/OpenTelemetry/Logs/BatchLogRecordExportProcessor.cs

```diff
@@ -37,6 +40,8 @@ public class BatchLogRecordExportProcessor : BatchExportProcessor<LogRecord>
 
         public override void OnEnd(LogRecord data)
         {
+            Guard.ThrowIfNull(data);
```

Comment:
Do we reallyneed this, or it can be a Debug.Assert?

Reply:
It is public so anyone is free to call it in strange ways. I think by convention we should validate anything passed to public methods. But practically speaking, probably never going to be null so we are just wasting cycles 😄 I'll switch it to `Debug.Assert` since it sounds like you are fine breaking with convention here.